File: partner_modules/sdk/python/vaultfire.py
"""Minimal Vaultfire Python SDK stub."""
import requests

def loyalty(user_id):
    return requests.get(f"/loyalty/{user_id}").json()

def identity(name):
    return requests.get(f"/resolve/{name}").json()

def reward(user, amount):
    return requests.post("/reward", json={"user": user, "amount": amount}).json()

File: partner_modules/sdk/js/vaultfire.js
export async function loyalty(userId) {
  return fetch(`/loyalty/${userId}`).then(r => r.json());
}
export async function identity(name) {
  return fetch(`/resolve/${name}`).then(r => r.json());
}
export async function quiz(data) {
  return fetch('/ns3/quiz', {method: 'POST', body: JSON.stringify(data)}).then(r => r.json());
}
export async function wallet(addr) {
  return fetch(`/wallet/${addr}`).then(r => r.json());
}
export async function reward(user, amount) {
  return fetch('/reward', {method: 'POST', body: JSON.stringify({user, amount})});
}
export async function signal(data) {
  return fetch('/signal', {method: 'POST', body: JSON.stringify(data)});
}
